subsys/caf/modules/buttons.c

```diff
@@ -179,8 +206,19 @@ static int suspend(void)
 		break;
 
 	case STATE_ACTIVE:
-		state = STATE_IDLE;
-		err = callback_ctrl(true);
+		/* Disable callbacks (and cancel the work) to ensure it will not be scheduled by an
+		 * invalid button in the idle state.
+		 */
+		err = callback_ctrl(MASK_NONE);
+
+		if (!err) {
+			/* Setup callbacks and columns for the idle state. */
+			state = STATE_IDLE;
+			err = set_cols(mask_wakeup(col, ARRAY_SIZE(col)), true);
```

Comment:
this is the only place when `set_cols` should have `true` in arguments

btw - can you create wakeup mask in temp variable for improved readability?
```
uint32_t wakeup_cols = get_wakeup_mask(col, ARRAY_SIZE(col));
uint32_t wakeup_rows = get_wakeup_mask(row, ARRAY_SIZE(row));

err = set_cols(wakeup_cols, true);
if (!err) {
   err = callback_ctrl(wakeup_rows)
}
```